cmake/modules/XROOTDApplePostInstall.cmake

```diff
@@ -0,0 +1,43 @@
+# Post install routine needed to change the LC_RPATH variable of the XRootD
+# libraries on macOS after installation. The function changes the value from
+# ${build_libdir} to ${install_libdir} with the `install_name_tool` executable.
+# On successive reiterations of the `cmake --install` command,
+# `install_name_tool` would error out since LC_RPATH would be already set to
```

Comment:
Why would it see the install RPATH? I'd expect that `make install` replaces `/usr/local/lib/libXrdFoo.so` by `${CMAKE_BUILD_DIR}/lib/libXrdFoo.so` which still has the build-time RPATH?

Reply:
It's a bit convoluted . The first time this [line](https://github.com/root-project/root/blob/b1b83e552c8418c74ccfb5c2828b4d8a9809e502/cmake/modules/SearchInstalledSoftware.cmake#L919) is executed the library would be moved from `${CMAKE_BUILD_DIR}/lib/libXrdFoo.so` to `${CMAKE_INSTALL_PREFIX}/lib/libXrdFoo.so`. Then this post-install script would call the following
```
install_name_tool -rpath ${CMAKE_BUILD_DIR}/lib ${CMAKE_INSTALL_PREFIX}/lib ${CMAKE_INSTALL_PREFIX}/lib/libXrdFoo.so
```
And this call would produce no errors since the current `LC_RPATH` variable for `libXrdFoo.so` is pointing to the build directory.

But then, anytime the user calls `make install` again, the routine above would be repeated with the difference that now `LC_RPATH` is already pointing to `${CMAKE_INSTALL_PREFIX}/lib `. Calling `install_name_tool` would then error out
```
error: install_name_tool: no LC_RPATH load command with path ${CMAKE_BUILD_DIR}/lib found in:
${CMAKE_INSTALL_PREFIX}/lib/libXrdFoo.so, required for specified option
"-rpath ${CMAKE_BUILD_DIR}/lib ${CMAKE_INSTALL_PREFIX}/lib"
```

And since we are happy with LC_RPATH pointing already to the install directory we can discard this type of error